### PR TITLE
Fix native_batch_norm_backward returning non-channels_last_3d grad

### LINKED ISSUE #107199: CUDA native_batch_norm_backward returns non-channels last grad for channels last input
### 🐛 Describe the bug

native_batch_norm_backward on CUDA returns non-channels last grad for channels last input, but it returns channels last grad on CPU.

```
# reproducer
import torch
import copy
import torch.utils._pytree as pytree

x = torch.randn(2, 3, 4, 4, 4).cuda().requires_grad_()
x2 = x.contiguous(memory_format=torch.channels_last_3d).cuda().detach().requires_grad_()
model = torch.nn.BatchNorm3d(3, 0.001, 0.7, False).cuda()
model2 = copy.deepcopy(model).to(memory_format=torch.preserve_format).cuda()
output = model(x)
grad_out = torch.randn(output.shape).cuda()
grad_out2 = grad_out.clone().contiguous(memory_format=torch.channels_last_3d).cuda()

output2 = model2(x2)

if isinstance(output2, torch.Tensor):
    outputs2 = (output2,)


# === Do backward pass. ===
diff_outputs = tuple(t for t in outputs2)
params = tuple(p for p in model2.parameters())
inputs=(x2,)
diff_inputs = tuple(
    t
    for t in pytree.tree_flatten((inputs, params))[0]
)
grad_outputs = tuple(
    torch.rand_like(t)
    for t in diff_outputs
)

grad_inputs = torch.autograd.grad(
    diff_outputs,
    diff_inputs,
    grad_outputs=grad_outputs,
    allow_unused=True,
)

for gi in grad_inputs:
    print("gi.shape: ", gi.shape, " gi.stride: ", gi.stride())
```

Run the reproducer: `gi.shape:  torch.Size([2, 3, 4, 4, 4])  gi.stride:  (192, 64, 16, 4, 1)`. The memory format of grad for input of `channels_last_3d` is channels first.

### Versions

Latest PyTorch.

cc @ezyang @gchanan @zou3519 @albanD @mruberry @jbschlosser @walterddr @mikaylagawarecki @ptrblck @jamesr66a

### PR DESCRIPTION
**Pull request by mikaylagawarecki**
Fix #107199

Checked out https://github.com/pytorch/pytorch/pull/106104 which caught this locally and verified that https://github.com/pytorch/pytorch/blob/551124f67090b7c672e9833ed10f41d31c57747c/torch/testing/_internal/common_modules.py#L2635-L2642 with the `p['device'] == 'cuda'` part shifted to `device_type = 'cuda'` now succeeds

Stack from [ghstack](https://github.com/ezyang/ghstack) (oldest at bottom):
* __->__ #107270

### CODE REVIEW

File: aten/src/ATen/native/cuda/Normalization.cu [hunk 1]
```diff
@@ -211,7 +211,7 @@ Tensor batch_norm_elementwise_backward_train(
     auto weight_nd = weight.defined() ? as_nd(weight) :
         at::scalar_tensor(1.0, input.options().dtype(mean.scalar_type()));
 
-    Tensor grad_input = at::empty(input.sizes(), grad_out.options());
+    Tensor grad_input = at::empty(input.sizes(), grad_out.options()).to(input.suggest_memory_format());
```
Comment (albanD) [inline aten/src/ATen/native/cuda/Normalization.cu:214]: add this to the options() argument of empty. No need for a full copy!

Comment (mikaylagawarecki) [reply aten/src/ATen/native/cuda/Normalization.cu]: my bad, fixed!

**Review by albanD — APPROVED** [review]
SGTM
